pkg/processor/sourceprocessor/source_category_filler.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2021 Sumo Logic, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sourceprocessor
+
+import (
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+type sourceCategoryFiller struct {
+	attributeName      string
+	valueTemplate      string
+	templateAttributes []string
+	prefix             string
+	dashReplacement    string
+	annotationPrefix   string
+}
+
+func newSourceCategoryFiller(
```

Comment:
Comment ? 😢

Reply:
Does this make sense? 

```go
// newSourceCategoryFiller creates a new sourceCategoryFiller.
```

Reply:
A bit more verbose? What does it fill?When?How?

Reply:
Does the `fill` function comment suffice or should I add more here?